cardinal/world_context.go

```diff
@@ -60,6 +63,12 @@ type worldContext struct {
 	txPool   *txpool.TxPool
 	logger   *zerolog.Logger
 	readOnly bool
+	rand     *rand.Rand
+}
+
+func createRngFromWorld(world *World) *rand.Rand {
+	//nolint:gosec // we require manual in the rng which crypto/rand doesn't have, but math/rand does.
+	return rand.New(rand.NewSource(int64(world.timestamp.Load())))
```

Comment:
nit: i would just inline this into `newWorldContextForTick`. The other world context should not have a rand object, see my previous comment.